dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/Utils.java

```diff
@@ -843,7 +843,7 @@ private Object wrapForEmbedding(HALResource<? extends RestAddressableModel> reso
         } else if (linkedObject instanceof List) {
             // The full list has been retrieved and we need to provide the first page for embedding
             List<RestAddressableModel> list = (List<RestAddressableModel>) linkedObject;
-            if (list.size() > 0) {
+            if (list != null && list.size() > 0) {
```

Comment:
LGTM says this is a "useless null check" because `list` can never be null "since it is guarded by `instanceof`" check above.  So this change should be removed.